### ISSUE TITLE
[Table]: WCAG and other issues

### ISSUE THREAD

**Issue by ivarkallejarv** [TEDI-Design-System/react#111]
## Problem
- Add sorting and/or filtering to table. JAWS reads out "Filtreeri, button" and "Sorteeri kasvavalt", which gives no indication as to what you can filter by, i.e. filter by name or etc etc (column name most likely, not 100% sure what the standard is here)
![Image](https://github.com/user-attachments/assets/e28f77cf-115a-4b75-97a7-5dbc12d1b0ed)

- Add items to table, and also filter if needed. Upon state changes, pagination doesn't read out number of results or other necessary information
![Image](https://github.com/user-attachments/assets/c18e79a7-62a2-4df2-ac0a-dd98c48c2902)

- Add 2 columns to table, display 2 results. Navigate to table using JAWS, it will read out for example - "Type, content of type, column 1", "Number, filtreeri sorteeri kasvavalt, content of number, column 2", etc..
Column heading should be read out, but sorting/filtering should not
